esp-hal/src/radio_clock_ctrl.rs

```diff
@@ -0,0 +1,171 @@
+//! Control over the radio clocs.
+use core::cell::RefCell;
+
+use critical_section::Mutex;
+use portable_atomic::{AtomicBool, Ordering};
+
+use crate::peripherals::RADIO_CLK;
+
+/// Enumeration of the available radio peripherals for this chip.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[cfg(any(bt, ieee802154, wifi))]
```

Comment:
If none of these symbols exist, but `phy` does for some reason, I think this module just won't compile. I'm thinking this cfg either doesn't need to exist, or it should be gating `pub mod radio_clock_ctrl`.

Reply:
I was unsure here too, because if a chip has any of the above, a PHY must be present. The only chip without a PHY is the P4. The module itself is feature gated.

Reply:
I think this cfg can go away. It's a minor thing, but if we just look at this symbolically (i.e., disregarding that phy (currently) implies one of these), I'd prefer not having a theoretical combination that fails to compile.

Reply:
Oh, I was talking about the wrong cfg. I have removed the one on line 12, since it doesn't do anything.